### DISCUSSION TITLE
Was this inspired by Recoil?

### DISCUSSION THREAD

**Discussion by transon-top** [pmndrs/jotai#1252] [Q&A] [answered]
It feels a lot like Recoil by facebookexperimental.

**Comment by AjaxSolutions** [comment]
The Jotai documentation mentions that the atomic model was inspired by Recoil.

After reading the documentation and playing with Jotai, one might get the impression that atoms are (a) good for tracking dependencies and (b) sharing global state between the components, but the atomic programming model is far more radical.

This is because the entire application logic can be stored in atoms and thus simplify the components by separating the business logic from the JSX presentation layer.

**Comment by dai-shi** [comment, marked as answer]
Some random notes:
- jotai was inspired by recoil.
- I had been developing various libraries to unlimit React Context pattern.
- atomic model is a good solution populated by recoil.
- recoil is nice but we didn't like the string key.
- atomic architecture is common between jotai and recoil, and maybe some others.
- There are many small differences between jotai and recoil. Their goals can be different too.

**Reply by transon-top** [reply]
thanks. jotai is a treasure level lib . I think I find it  : )
